### PR TITLE
[DOCS] Update URLs in plugin document

### PR DESCRIPTION
**Pull request by GabyCT**
This PR updates the URLs for several references that are being
used in the plugin document.

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: docs/plugins/authors.asciidoc [hunk 1]
```diff
@@ -3,12 +3,12 @@
 
 :plugin-properties-files: {elasticsearch-root}/build-tools/src/main/resources
 
-The Elasticsearch repository contains https://github.com/elastic/elasticsearch/tree/master/plugins/examples[examples of plugins]. Some of these include:
+The Elasticsearch repository contains https://github.com/elastic/elasticsearch/tree/main/plugins/examples[examples of plugins]. Some of these include:
```
Comment (abdonpijpelink) [inline docs/plugins/authors.asciidoc:6]:
```suggestion
The Elasticsearch repository contains {es-repo}tree/main/plugins/examples[examples of plugins]. Some of these include:
```

File: docs/plugins/authors.asciidoc [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
 
-* a plugin with https://github.com/elastic/elasticsearch/tree/master/plugins/examples/custom-settings[custom settings]
-* adding https://github.com/elastic/elasticsearch/tree/master/plugins/examples/rest-handler[custom rest endpoints]
-* adding a https://github.com/elastic/elasticsearch/tree/master/plugins/examples/rescore[custom rescorer]
-* a script https://github.com/elastic/elasticsearch/tree/master/plugins/examples/script-expert-scoring[implemented in Java]
+* a plugin with https://github.com/elastic/elasticsearch/tree/main/plugins/examples/custom-settings[custom settings]
+* adding https://github.com/elastic/elasticsearch/tree/main/plugins/examples/rest-handler[custom rest endpoints]
+* adding a https://github.com/elastic/elasticsearch/tree/main/plugins/examples/rescore[custom rescorer]
+* a script https://github.com/elastic/elasticsearch/tree/main/plugins/examples/script-expert-scoring[implemented in Java]
```
Comment (abdonpijpelink) [inline docs/plugins/authors.asciidoc:11]:
```suggestion
* a plugin with {es-repo}tree/main/plugins/examples/custom-settings[custom settings]
* adding {es-repo}tree/main/plugins/examples/rest-handler[custom rest endpoints]
* adding a {es-repo}tree/main/plugins/examples/rescore[custom rescorer]
* a script {es-repo}tree/main/plugins/examples/script-expert-scoring[implemented in Java]
```

**Review by abdonpijpelink — APPROVED** [review]
Thanks @GabyCT ! I've left a couple of minor suggestions to also use the `{es-repo}` variable.

File: docs/plugins/authors.asciidoc (hunk 2 above)
Comment (GabyCT) [reply docs/plugins/authors.asciidoc]: @abdonpijpelink thanks for the feedback changes applied

File: docs/plugins/authors.asciidoc (hunk 1 above)
Comment (GabyCT) [reply docs/plugins/authors.asciidoc]: @abdonpijpelink thanks for the feedback changes applied

**Pull request merged**